tests/sentry/demo/test_demo_start.py

```diff
@@ -1,31 +1,74 @@
-from sentry.utils.compat import mock
-
 from django.test.utils import override_settings
+from django.core import signing
 from exam import fixture
 
+from sentry.demo.demo_start import MEMBER_ID_COOKIE
+from sentry.models import OrganizationStatus
 from sentry.testutils import TestCase
+from sentry.utils.compat import mock
 
+signer = signing.get_cookie_signer(salt=MEMBER_ID_COOKIE)
 
-class AuthLoginTest(TestCase):
+
+@override_settings(DEMO_MODE=True, ROOT_URLCONF="sentry.demo.urls")
+class DemoStartTeset(TestCase):
     @fixture
     def path(self):
         return "/demo/start/"
 
-    @override_settings(DEMO_MODE=True, ROOT_URLCONF="sentry.demo.urls")
+    def setUp(self):
+        super().setUp()
+        self.user = self.create_user()
+        self.org = self.create_organization()
+        self.member = self.create_member(organization=self.org, role="member", user=self.user)
+
     @mock.patch("sentry.demo.demo_start.auth.login")
     @mock.patch("sentry.demo.demo_org_manager.assign_demo_org")
     def test_basic(self, mock_assign_demo_org, mock_auth_login):
-
-        user = self.create_user()
-        org = self.create_organization()
-
-        mock_assign_demo_org.return_value = (org, user)
+        mock_assign_demo_org.return_value = (self.org, self.user)
         resp = self.client.post(self.path)
         assert resp.status_code == 302
-
-        mock_auth_login.assert_called_once_with(mock.ANY, user)
+        mock_auth_login.assert_called_once_with(mock.ANY, self.user)
+        recovered = resp.cookies[MEMBER_ID_COOKIE].value.split(":")[0]
+        assert recovered == str(self.member.id)
 
     @override_settings(DEMO_MODE=False, ROOT_URLCONF="sentry.demo.urls")
     def test_disabled(self):
         resp = self.client.post(self.path)
         assert resp.status_code == 404
+
+    @mock.patch("sentry.demo.demo_start.auth.login")
+    def test_member_cookie(self, mock_auth_login):
+        self.save_cookie(MEMBER_ID_COOKIE, signer.sign(self.member.id))
+        resp = self.client.post(self.path)
+        assert resp.status_code == 302
```

Comment:
Do you care which URL the user is being sent to here?

Reply:
@markstory I have other tests for that spec